src/handles/handleFormat.ts

```diff
@@ -38,7 +47,11 @@ async function handleFormat(
     }
   )
   let output = '';
-  if (isStdout === undefined && isStderr === undefined) {
+  if (code > 0) {
+    output = text
+  } else if (config.doesWriteToFile) {
+    output = fs.readFileSync(VscUri.parse(textDocument.uri).fsPath, 'utf8')
```

Comment:
This is to support formatters that don't support stdin / stdout. Granted, it's not much use in coc.nvim due to its small timeout for formatters.

Reply:
I could also remove `doesWriteToFile` and instead check for `isStdout === false && isStderr === false`